### DISCUSSION TITLE
Pixel Aspect Ratio Correction?

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#2722] [General]
Hi everyone!  I hope you are well and safe during this craziness.  : )

I'm wondering if there's anyway to apply PAR correction when resizing, for IM to recognize the PAR of a source image and create a kind of scaled visualization of it when reformatting?

In my source files, the only metadata I see that holds that information is _photoshop:pixelaspectratio_ but presumably this would be stored or discerned another way.  The attached sample file is a black JPEG with anamorphic (2:1) PAR, and would appear like a square without PAR correction, and like a rectangle with 2x width with scaling applied. I want to make another image of this automatically that emulates the PAR correction, so using convert basically to make a new JPEG that is a rectangle to visualize the PAR, and would work with varying PARs.

Thanks in advance for your time in considering my question!  - J

![sample_anamorphic_PAR](https://user-images.githubusercontent.com/3990142/96062413-79dfa200-0e4a-11eb-97e0-af7f8436a209.jpg)

**Comment by fmw42** [comment]
Please provide a reference about PAR.

**Reply by blue-j** [reply]
> Please provide a reference about PAR.

This is an example of PAR correction being applied to a preview in Photoshop:

<img width="849" alt="Screen Shot 2020-10-14 at 6 58 12 PM" src="https://user-images.githubusercontent.com/3990142/96067741-46534680-0e4f-11eb-9486-9947c702e7ee.png">

**Comment by blue-j** [comment]
Hi Fred!  I tried to! The black square is an example of a JPEG with a PAR of 2.  With correction it should look like a rectangle, with a width 2x what is shown, and a height 1/2 of what is shown.

**Reply by fmw42** [reply]
Looking for theoretical or algorithmic reference as background and definition

**Reply by blue-j** [reply]
https://en.wikipedia.org/wiki/Pixel_aspect_ratio

**Reply by blue-j** [reply]
You see it most in video, but still images, especially when being prepared for use in said videos, use varying PARs as well:

<img width="604" alt="Screen Shot 2020-10-14 at 7 01 21 PM" src="https://user-images.githubusercontent.com/3990142/96067968-b19d1880-0e4f-11eb-99c9-a643cac55ade.png">

**Reply by fmw42** [reply]
How about:

`magick image -resize 2:1 result`

Or are you asking to go back the other way. If so, that information is lost in the pixel data once it has been resized. That information would need to be stored in the meta data of the file.

However, if you know it had been scaled 2:1. You could then correct by

`magick image -resize 50x100% result`

or

`magick image -resize 0.5x1 result`

**Reply by blue-j** [reply]
Thanks for the suggestion, but I am trying to deal with automatic batch processing of unknown PARs.  Maybe we have to do it in intelligence outside of IM - grab the metadata and apply it in the resize as a variable?

**Comment by snibgo** [comment]
exiftool says:
```
Slices Group Name               : sample_anamorphic_PAR
Num Slices                      : 1
Pixel Aspect Ratio              : 2
```
From that, I assume you know it needs stretching by a factor of 2 vertically (or perhaps horizontally). I would use "-resize 100x200%", but perhaps "-scale" or "-sample" is more appropriate.

Confusingly, exiftool also says:
```
X Resolution                    : 144
Y Resolution                    : 144
```
I would expect these numbers to be different.

**Reply by blue-j** [reply]
Thanks so much to both of your for your time in reflecting on my post.  I have concluded that we will set this logic outside of IM, which is totally fine.  Thanks again!   - J

**Reply by t246246** [reply]
You probably already read 0x0428 part of https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/

(Photoshop CS) Pixel Aspect Ratio. 4 bytes (version = 1 or 2), 8 bytes double, x / y of a pixel. Version 2, attempting to correct values for NTSC and PAL, previously off by a factor of approx. 5%.

I don't understand version 2 comment, though. At quick look at exiftool source, https://github.com/paleolimbot/exifr/blob/master/inst/exiftool/lib/Image/ExifTool/Photoshop.pm it seems to just print real value part.
